pkg/v1/tkg/aws/client.go

```diff
@@ -372,8 +374,9 @@ func (c *client) ListInstanceTypes(optionalAZName string) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
-		filteredInstances := filterInstanceType(candidates)
-		diffInstancesPerAz := getSetDifference(instances, filteredInstances)
+		// filter and build a list of unsupported instance types
+		unsupportedInstances := getUnsupportedInstanceTypes(candidates)
```

Comment:
Nothing blocking here, but just curious why we don't have one call here of `getSupportedInstanceTypes` rather than getting the unsupported and then filtering after that.

Reply:
Ah, nevermind. I've read through the following code a little closer and see there are multiple conditions where we need to filter out the unsupported instances from different sets of possible instance types.